ComponentKit/Core/CKComponentController.mm

```diff
@@ -142,32 +164,36 @@ - (void)componentDidMount:(CKComponent *)component
     case CKComponentControllerStateMounting:
       _state = CKComponentControllerStateMounted;
       [self didMount];
-      for (const auto &pendingAnimation : _pendingAnimationsOnInitialMount) {
-        const CKComponentAnimation &anim = pendingAnimation.animation;
-        [CATransaction begin];
-        CKComponentAnimationID animationID = _nextAnimationID++;
-        [CATransaction setCompletionBlock:^() {
-          eraseAnimation(_appliedAnimationsOnInitialMount, animationID);
-        }];
-        _appliedAnimationsOnInitialMount.insert({animationID, {anim, anim.didRemount(pendingAnimation.context)}});
-        [CATransaction commit];
+      if (_animationData) {
+        for (const auto &pendingAnimation : _animationData->pendingAnimationsOnInitialMount) {
+          const CKComponentAnimation &anim = pendingAnimation.animation;
+          [CATransaction begin];
+          CKComponentAnimationID animationID = _animationData->nextAnimationID++;
+          [CATransaction setCompletionBlock:^() {
+            eraseAnimation(_animationData->appliedAnimationsOnInitialMount, animationID);
+          }];
+          _animationData->appliedAnimationsOnInitialMount.insert({animationID, {anim, anim.didRemount(pendingAnimation.context)}});
+          [CATransaction commit];
+        }
+        _animationData->pendingAnimationsOnInitialMount.clear();
       }
-      _pendingAnimationsOnInitialMount.clear();
       break;
     case CKComponentControllerStateRemounting:
       _state = CKComponentControllerStateMounted;
       [self didRemount];
-      for (const auto &pendingAnimation : _pendingAnimations) {
-        const CKComponentAnimation &anim = pendingAnimation.animation;
-        [CATransaction begin];
-        CKComponentAnimationID animationID = _nextAnimationID++;
-        [CATransaction setCompletionBlock:^() {
-          eraseAnimation(_appliedAnimations, animationID);
-        }];
-        _appliedAnimations.insert({animationID, {anim, anim.didRemount(pendingAnimation.context)}});
-        [CATransaction commit];
+      if (_animationData) {
+        for (const auto &pendingAnimation : _animationData->pendingAnimations) {
+          const CKComponentAnimation &anim = pendingAnimation.animation;
+          [CATransaction begin];
+          CKComponentAnimationID animationID = _animationData->nextAnimationID++;
+          [CATransaction setCompletionBlock:^() {
+            eraseAnimation(_animationData->appliedAnimations, animationID);
+          }];
+          _animationData->appliedAnimations.insert({animationID, {anim, anim.didRemount(pendingAnimation.context)}});
+          [CATransaction commit];
+        }
+        _animationData->pendingAnimations.clear();
```

Comment:
It's a bit of a bummer that this block of code is so similar to the one above. Seems like a potential place for behavior to get out of sync. Likely not a concern of this change but maybe something we can clean up separately?